src/sidebar/services/groups.js

```diff
@@ -163,164 +163,144 @@ function groups(
    *
    * @return {Promise<Group[]>}
    */
-  function load() {
-    let uri = Promise.resolve(null);
+  async function load() {
+    // Step 1: Get the URI of the active document, so we can fetch groups
+    // associated with that document.
+    let uri = null;
```

Comment:
I could be wrong but I think this code could be simplified by doing:
```
let documentUri = null;
.
.
.
    if (isSidebar) {
      documentUri = await getDocumentUriForGroupSearch();
    }
```

Reply:
You're right. I'll do this.